README.md

```diff
@@ -8,79 +8,211 @@ JavaScript Closures
+the `whatsForDinner()` function at a later point in time, when we're curious about what exactly is in Fat Bastard's belly.
 
-- Feel free to reuse the sandwich metaphor as much as you'd like
+The feature allowing `whatsForDinner()` to still access the variables within its parent function long after the parent
+function has executed, is called a 'closure'. In this case, the `fatBastard()` function is a closure.
 
-When we say, "A closure closes over its environment," we mean that it keeps track of the variables it had access to when it was created.
+![Fat Bastard in a moment of spiritual clarity.](https://66.media.tumblr.com/11911f29151cb0ff4813f9d11689cdc8/tumblr_mvjxzdTtV91ror07qo1_500.gif)
 
-- Walk through simple example, perhaps based on the first one at MDN:
+To prove that Fat Bastard can still let us know what is in his belly after he's eaten it, let's feed him a nice, juicy
+steak.
 
-``` javascript
-function init() {
-  var name = "Chuck"
-  function logName() {
-    console.log(name)
+```js
+const whatsForDinner = fatBastard('Kobe beef');
+whatsForDinner(); // prints 'I'm eatin' a bit of Kobe beef! Burp.'
+```
+
+Keep in mind that since we're returning a function, there's no need to use the same name for our variable as the function
+being returned. This would work too:
+
+```js
+const whatsInHisTummy = fatBastard('Mini-Me');
+whatsInHisTummy(); // prints 'The wee man is in my belly!'
+```
+
+## Gotchas
+
+The environment of a closure is remembered and not discarded. That means that the variables in that scope are shared
+(just like regular variables), and can also be changed. This is something you need to be aware of. To illustrate, let's
+expand on what Fat Bastard can do with his meal, by allowing him to digest it:
+
+```js
+function fatBastard(meal) {
+  function whatsForDinner() {
+    if (!meal) {
+      console.log('My belly is empty. Woe is me.');
+    } else if (meal === 'Mini-Me') {
+      console.log('The wee man is in my belly!');
+    } else {
+      console.log(`I'm eatin' a bit of ${meal}! Burp.`);
+    }
   }
-  logName()
+
+  function digest() {
+    meal = undefined;
+  }
+
+  return {
+    whatsForDinner,
+    digest
+  };
 }
+```
+
+Keep in mind that we're now returning **an object** with references to our `whatsForDinner()` and `digest()` functions,
+meaning that now we **do** need to use the key names to refer to them! Let's serve him another slab of beef, and give him
+some time to digest it:
 
-init()
+```js
+const { whatsForDinner, digest } = fatBastard('ribeye');
+whatsForDinner(); // prints 'I'm eatin' a bit of ribeye! Burp.'
+digest();
+whatsForDinner(); // prints 'My belly is empty. Woe is me.'
 ```
 
-## How does a closure work?
+Another thing to watch out for is that closures are the most common source of performance issues and memory leaks. Since
+the variables that are closed over might still be in use, they're either never or barely picked up by the garbage
```

Comment:
It might be worth an aside on what "garbage collection" is (no need to go super in-depth).